lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3044,6 +3054,16 @@ namespace {
       result->setIsObjC(false);
       result->setIsDynamic(false);
 
+      bool isFromFriend = Impl.cxxFriends.contains(decl);
```

Comment:
Can we put this in the `importFunctionDecl` method? Can we just look at the decl context instead of `Impl.cxxFriends`.

Reply:
Cant look at `declContext` because the passed in decl is `clang::FunctionDecl` and its context is not `clang::FriendDecl` but rather `TranslationUnit`.

Reply:
The aptly named `clang::FunctionDecl:: isThisDeclarationInstantiatedFromAFriendDefinition` :)

Reply:
Whhhahaaaattttt 🤯